### PR TITLE
implement CURLOPT_CURLU

### PR DESCRIPTION
**Pull request by xquery**
Implement CURLOPT_CURLU (as per https://github.com/curl/curl/wiki/URL-API) so one may pass via curl_easy_setopt aka:

  CURL *handle;
  CURLU *urlp;
  CURLUcode uc;
  
  urlp = curl_url();
  uc = curl_url_set(urlp, CURLUPART_URL, "http://example.com/path/index.html", 0);

  curl_easy_setopt(handle, CURLOPT_CURLU, urlp);
  ...

this approach expands on the prior attempts ( #3128, #3160,#3188) ... updated docs (CURLOPT_CURLU.3) to highlight override behavior as well as adding demonstration of this in libtest658.c.

ps - yes, git push --force is now my friend

### CODE REVIEW

**Comment by xquery** [comment]
I think appveyor borked due to other reasons ...

File: docs/libcurl/opts/CURLOPT_CURLU.3 [hunk 1]
```diff
@@ -29,7 +29,7 @@ CURLOPT_CURLU \- set URL with CURLU *
 CURLcode curl_easy_setopt(CURL *handle, CURLOPT_CURLU, void *pointer);
 .SH DESCRIPTION
 Pass in a pointer to the \fIURL\fP to work with. The parameter should be a
-CURLU *.
+CURLU *. Setting CURLOPT_CURLU will explicitly override CURLOPT_URL.
```
Comment (bagder) [inline docs/libcurl/opts/CURLOPT_CURLU.3]: if you write them as `\fICURLOPT_CURLU(3)\fP` and `\fICURLOPT_URL(3)\fP` they will appear suitably highlighted in man pages and hyperlinked on the web pages.

File: docs/libcurl/opts/CURLOPT_CURLU.3 [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  curl_url_cleanup(urlp);
+  curl_easy_cleanup(handle);
+}
+.fi
+.SH AVAILABILITY
+Added in 7.63.0.
+.SH RETURN VALUE
+Returns CURLE_OK if the option is supported, and CURLE_UNKNOWN_OPTION if not.
+.SH "SEE ALSO"
+.BR CURLOPT_URL "(3),
```
Comment (bagder) [inline docs/libcurl/opts/CURLOPT_CURLU.3]: maybe also some of the URL API functions can be mentioned in the SEE ALSO?

**Comment by bagder** [comment]
If you rebase and force-push, you can get the merge conflict resolved and all test should be able to run green now.

File: docs/libcurl/opts/CURLOPT_CURLU.3 (hunk 2 above)
Comment (xquery) [reply docs/libcurl/opts/CURLOPT_CURLU.3]: ack

File: docs/libcurl/opts/CURLOPT_CURLU.3 (hunk 1 above)
Comment (xquery) [reply docs/libcurl/opts/CURLOPT_CURLU.3]: ack

**Comment by moteus** [comment]
Can you please specify in doc Is it possible change URL after CULR_URLU was set.
So make multible requests but instead of reset URL on easy handle change it only in url handle

**Comment by xquery** [comment]
> Can you please specify in doc Is it possible change URL after CULR_URLU was set.
> So make multible requests but instead of reset URL on easy handle change it only in url handle

seems reasonable, though will be the subject of a different PR if we do it.

**Comment by bagder** [comment]
I think that feature is implied but maybe we should spell it out better. And possibly also spell out that the handle is used read-only by libcurl itself...

**Pull request closed without merging**